### ISSUE TITLE
Removal of #[\ReturnTypeWillChange] Attribute by Mozart After Scooping

### ISSUE THREAD

**Issue by mostafasoufi** [coenjacobs/mozart#146]
It appears that after running the Mozart package, the `#[\ReturnTypeWillChange]` attribute is being removed from the methods in my code.

This attribute is important for maintaining compatibility with certain PHP versions and ensuring the proper functioning of my methods.

Any idea?

**Comment by coenjacobs** [comment]
@mostafasoufi Do you have an example of a package that includes this attribute, that it is removing it from? Or do you use it in your own code and Mozart removes this for you? Can you give me some more context please?

**Comment by mostafasoufi** [comment]
Yep! Take a look at this repository https://github.com/wp-statistics/wp-statistics

PS. I added the attribute `#[\ReturnTypeWillChange]` manually after running `mozart compose`

**Comment by coenjacobs** [comment]
This was caused by the regex-based replacement engine, which operated on raw text and could inadvertently corrupt PHP syntax constructs it wasn't designed to handle, including PHP 8.0+ attributes like `#[\ReturnTypeWillChange]`.

On the current `master` branch (upcoming 1.1.0 release), the entire replacement engine has been rewritten to use AST-based processing via `nikic/php-parser`. The AST approach parses PHP code into a proper syntax tree, modifies only the specific nodes that need prefixing (namespace declarations, use statements, and class name references), and then prints the modified tree back to code. PHP attributes are preserved as-is since they are separate node types that are never targeted for modification. This will be available starting with the first 1.1.0 pre-release.

**Comment by coenjacobs** [comment]
The [first beta of the upcoming 1.1.0 release](https://github.com/coenjacobs/mozart/discussions/208) is now available. It would be really helpful if this specific scenario could be tested with this new version, to prove it has been fixed, in a real world scenario. 🙌
